build.py

```diff
@@ -359,6 +362,99 @@ def RunMakeZipArchive(context):
                     PrintError("Failed to write to directory {pkgDir} : {exp}".format(pkgDir=pkgDir,exp=exp))
                     sys.exit(1)
 
+def SetupMayaQt(context):
+    def haveQtHeaders(rootPath):
+        if os.path.exists(rootPath):
+            # MayaUsd uses these components from Qt (so at a minimum we must find them).
+            qtComponentsToFind = ['QtCore', 'QtGui', 'QtWidgets']
+            # Qt6 includes the entire Qt in a single zip file, which when extracted ends in folder 'Qt'.
+            startDir = os.path.join(rootPath, 'Qt', 'include') if os.path.exists(os.path.join(rootPath, 'Qt')) else os.path.join(rootPath, 'include')
+            for root,dirs,files in os.walk(startDir):
+                if 'qt' not in root.lower() or not files:
+                    continue
+                if not any(root.endswith(qtComp) for qtComp in qtComponentsToFind):
+                    # Skip any folders that aren't the components we are looking for.
+                    continue
+
+                for qtComp in qtComponentsToFind[:]:    # Loop over slice copy as we remove items
+                    if qtComp in root and '{comp}version.h'.format(comp=qtComp.lower()) in files:
+                        qtComponentsToFind.remove(qtComp)
+                        PrintInfo('Found {comp} in {dir}'.format(comp=qtComp, dir=root))
+                        break   # Once we've found (and removed) a component, move to the next os.walk
+
+                if not qtComponentsToFind:  # Once we've found them all, we are done.
+                    return True
+
+    # The list of directories (in order) that we'll search. This list matches the one
+    # in FindMayaQt.cmake.
+    dirsToSearch = [context.devkitLocation]
+    if 'MAYA_DEVKIT_LOCATION' in os.environ:
+        dirsToSearch.append(os.path.expandvars('$MAYA_DEVKIT_LOCATION'))
+    dirsToSearch.append(context.mayaLocation)
+    if 'MAYA_LOCATION' in os.environ:
+        dirsToSearch.append(os.path.expandvars('$MAYA_LOCATION'))
+
+    # Check if the Qt zip file has been extracted (we need the Qt headers).
+    for dirToSearch in dirsToSearch:
+        if haveQtHeaders(dirToSearch):
+            PrintStatus('Found Maya Qt headers in: {dir}'.format(dir=dirToSearch))
+            return
```

Comment:
Search thru all the locations and try to find the Qt headers, if we do return - no extra step needed.